### PR TITLE
docs: add Gatsby Integration and Gatsby Cloud Redirects and Rewrites

### PR DESCRIPTION
**Pull request by PaulieScanlon**

### CODE REVIEW

File: docs/gatsby.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+## Copy Library Files
+
+Copy library files to `static/~partytown`. How the files are copied or served from your site is up to each site's setup. A `partytown copylib` CLI [copy task](/copy-library-files) has been provided for convenience which helps copy the Partytown library files to the correct directory. Below is an example of using Gatsby's `onPreBuild` extension point. This can be added to `gatsby-node.js`.
+
+```jsx
+const path = require("path");
+
+exports.onPreBuild = async ({ actions: { createRedirect } }) => {
+  const { copyLibFiles } = await import("@builder.io/partytown/utils");
```
Comment (adamdbradley) [inline docs/gatsby.md:52]: Out of curiosity, why was this partytown utils a dynamic import here? If `gatsby-node.js` iis a commonjs file, are you able to import it at the top with 

```
const { copyLibFiles } = require('@builder.io/partytown/utils');
```

If there was an issue importing that util in commonjs that's something we should fix. If it's personal preference that's cool too.

File: docs/gatsby.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  copyLibFiles(path.join(__dirname, "static", "~partytown"));
```
Comment (adamdbradley) [inline docs/gatsby.md:53]: Should probably prefix with `async` just to be safe and ensure the files fully copy.

**Review by adamdbradley — CHANGES_REQUESTED** [review]

File: docs/integrations.md [hunk 3]
```diff
@@ -11,5 +11,6 @@ Partytown can work with any HTML page, and doesn't require a specific framework.
 - [React](/react)
 - [Remix](/remix)
 - [Shopify Hydrogen](/shopify-hydrogen)
+- [Gatsby](/gatsby)
```
Comment (adamdbradley) [inline docs/integrations.md:14]: Would you be able to move this under `Astro` so it's in alphabetical order?

File: docs/proxying-requests.md [hunk 4]
```diff
@@ -53,6 +53,7 @@ Below are a few examples of [reverse proxies](https://en.wikipedia.org/wiki/Reve
 - [Netlify: Proxy to another service](https://docs.netlify.com/routing/redirects/rewrites-proxies/#proxy-to-another-service)
 - [NGINX](https://docs.nginx.com/nginx/admin-guide/web-server/reverse-proxy/)
 - [Vercel: Rewrites](https://vercel.com/docs/cli#project-configuration/rewrites)
+- [Gatsby Cloud: Redirects and Rewrites](https://support.gatsbyjs.com/hc/en-us/articles/1500003051241-Working-with-Redirects)
```
Comment (adamdbradley) [inline docs/proxying-requests.md:56]: Same, can you move this up so it's alphabetical?

**Comment by adamdbradley** [comment]
Awesome thanks!!

**Comment by PaulieScanlon** [comment]
Thanks @adamdbradley! I think I might have missed one thing. There's a nav link that might need to be added. I'll have a look when I'm back at my computer. Ok if I create a second PR?

**Pull request merged**